cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,178 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring course "apps" such as progress, wiki, teams, discussions,
+etc. all from one place. Currently, these apps are intermingled with the
+core platform and are inseparable from it, however, future apps could be
+implemented as plugins.
+
+This ADR proposes a way to configure these apps from the backend that MFEs
+can use and potentially enable/disable these apps using an API.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can introduce a new type of Open edX plugin, a course app. These are
+course-level apps that add new functionality to the course for students.
```

Comment:
the word `app` is causing confusion here as if they are apps within other apps or what. 
Why are these called apps rather than plugins or anything else? Using `app` the hierarchy and the naming convention is causing confusion. 

e.g. Calling `TeamsCourseApp` that resides inside a Django app `team` at `lms/djangoapps/teams` is not easy to understand.
Compare this with something like, `TeamPlugin` inside Django app `team` is easier. I am just using `plugin` to reference for this example, this can be any other word that makes more sense 

This will help me get more context.
* I would suggest we should add a little detail why we are calling them `apps`? 
* Is there a specific reason for calling them apps?

Reply:
I don't think using the name plugin is good idea since that is too generic. There are already many different kinds of plugins. 

I've called them apps because that is how they have been referred to from the beginning.  For instance, here: https://openedx.atlassian.net/browse/TNL-7134

I don't think App is inherently confusing. An App is something that provides some bits of functionality, in that context a Course App is a something that provides a bit of functionality to a course. 

For instance, the edxnotes functionality is already referred to as the edxnotes app. The teams functionality is also called the teams app. The idea here is to make that official and provide a consistent way to find what are already called and considered apps, and return them via an API.